File: self/S2_07052025/ConvertPython/Ex2_3.m
%Ex2_3 Brightness Adjustment
clear, clc, close all;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%load test image
Img = im2double(imread('rose.jpg'));

%brightness adjustment by an intensity scaling
scale = 0.75; %scale=1: unchanged; scale>1: brighter; scale<1: darker
scaled_Img = Img .* scale;

%plot image
FS = 15; %fontsize of caption
figure, clf;
subplot(1,2,1), imshow(Img, [0 1]), title('Original Image', "fontsize", FS);
subplot(1,2,2), imshow(scaled_Img, [0 1]), title('Darker Image', "fontsize", FS);

%save image
saveas(gcf, 'Darker.jpg');

File: self/S2_07052025/ConvertPython/Ex2_Python/Ex2_3.py
import cv2
import matplotlib.pyplot as plt

img = cv2.imread('rose.jpg') / 255.0
scale = 0.5
scaled_img = img * scale
plt.figure(figsize=(8, 4))
plt.subplot(1, 2, 1), plt.imshow(cv2.cvtColor((img * 255).astype('uint8'), cv2.COLOR_BGR2RGB)), plt.title('Original Image', fontsize=15), plt.axis('off')
plt.subplot(1, 2, 2), plt.imshow(cv2.cvtColor((scaled_img * 255).astype('uint8'), cv2.COLOR_BGR2RGB)), plt.title('Brighter Image', fontsize=15), plt.axis('off')
plt.show()
plt.savefig('rose_darker.jpg')